digitalidentity/service.go

```diff
@@ -292,3 +279,36 @@ func GetShareReceipt(httpClient requests.HttpClient, receiptId string, clientSdk
 		Error: receiptResponse.Error,
 	}, nil
 }
+
+func validateReceiptResponse(receiptResponse ReceiptResponse) error {
+	if receiptResponse.Content == nil || len(receiptResponse.Content.Profile) == 0 {
+		return errors.New("received unexpectedly empty content or profile")
+	}
+	return nil
+}
+
+func decryptReceiptContent(content *Content, key []byte) (attrData *yotiprotoattr.AttributeList, aextra []byte, err error) {
+	aattr := []byte{}
+	if content != nil {
+		if len(content.Profile) > 0 {
+			aattr, err = cryptoutil.DecryptReceiptContent(content.Profile, key)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to decrypt receipt content profile: %v", err)
+			}
+		}
+
+		if len(content.ExtraData) > 0 {
+			aextra, err = cryptoutil.DecryptReceiptContent(content.ExtraData, key)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to decrypt receipt content extra data: %v", err)
+			}
+		}
+
+		attrData = &yotiprotoattr.AttributeList{}
+		if err := proto.Unmarshal(aattr, attrData); err != nil {
+			return nil, nil, fmt.Errorf("failed to unmarshal attribute list: %v", err)
+		}
```

Comment:
we need to move this unmarshalling to be above in the `if len(content.Profile) > 0` block (same with `aatr := []byte{}` var declaration)

Reply:
updated